File: backend/src/main/kotlin/com/back/back9/domain/orders/dto/OrderResponse.java
package com.back.back9.domain.orders.dto;

import com.back.back9.domain.coin.entity.Coin;
import com.back.back9.domain.orders.entity.Orders;

import java.math.BigDecimal;

public record OrderResponse(
        Long coinId,
        String coinSymbol,
    String coinName,
    String orderMethod,
    String orderStatus,
    String tradeType,
    BigDecimal price,
    BigDecimal quantity,
    String createdAt
) {
    public static OrderResponse from(Orders order) {
        Coin coin = order.getCoin();
        return new OrderResponse(
                coin.getId(),
                coin.getSymbol(),
                coin.getKoreanName(),
                order.getOrdersMethod().name(),
                order.getOrdersStatus().name(),
                order.getTradeType().name(),
                order.getPrice(),
                order.getQuantity(),
                order.getCreatedAt().toString()
        );
    }
}

File: backend/src/main/kotlin/com/back/back9/domain/orders/dto/OrderResponse.kt
package com.back.back9.domain.orders.dto

import com.back.back9.domain.orders.entity.Orders
import java.math.BigDecimal

@JvmRecord
data class OrderResponse(
    val coinId: Long?,
    val coinSymbol: String?,
    val coinName: String?,
    val orderMethod: String?,
    val orderStatus: String?,
    val tradeType: String?,
    val price: BigDecimal?,
    val quantity: BigDecimal?,
    val createdAt: String?
) {
    companion object {
        fun from(order: Orders): OrderResponse {
            val coin = order.getCoin()
            return OrderResponse(
                coin.getId(),
                coin.symbol,
                coin.koreanName,
                order.getOrdersMethod().name,
                order.getOrdersStatus().name,
                order.getTradeType().name,
                order.getPrice(),
                order.getQuantity(),
                order.getCreatedAt().toString()
            )
        }
    }
}
